yogstation/code/modules/power/nuclear_engine.dm

```diff
@@ -0,0 +1,449 @@
+		monitor.icon_state = "monitor-high"
+	if(Heat > CRITICAL_HEAT) //At this stage. Worry. A lot.
+		monitor.icon_state = "monitor-warning"
+	add_overlay(waste)
+	add_overlay(fuel)
+	add_overlay(monitor)
+
+/obj/machinery/power/NuclearReactor/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/twohanded/required/FuelRod))
+		if(state & FUELHATCH_OPEN)
+			to_chat(user, "You slot [I] into [src]!")
+			I.forceMove(src)
+			react.FRS += I
+		else
+			to_chat(user, "Open the fuel hatch first!")
+	if(istype(I, /obj/item/twohanded/required/ControlRod))
+		to_chat(user, "You slot [I] into [src]!")
+		I.forceMove(src)
+		react.CRS += I
+	if(istype(I, /obj/item/wrench))
+		CheckPipes()
+
+/obj/machinery/power/NuclearReactor/attack_hand(mob/user)
+	if(!ReactorInoperable)
+		if(state & WASTEHATCH_OPEN)
+			var/obj/C = input(user, "Remove what?", "Reactor Control Hatch") as null|obj in contents
+			C.forceMove(get_turf(user))
+			to_chat(user, "You carefully slide [C] out of [src]")
+			if(istype(C,/obj/item/twohanded/required/FuelRod))
+				react.FRS -= C
+			if(istype(C,/obj/item/twohanded/required/ControlRod))
+				react.CRS -= C
+			if(prob(20))
+				radiation_pulse(src, 5, 10)
+		else
+			start()
+	else
+		radiation_pulse(src, 100, 20)
+
+/obj/machinery/power/NuclearReactor/Initialize()
+	. = ..()
+	react.reactor = src
+	UpdateIcon()
+	state |= FUELHATCH_OPEN
+	state |= WASTEHATCH_OPEN
+	connect_to_network()
+//	outlet = new(src.loc)
+	outlet = locate(/obj/machinery/atmospherics/components/binary/pump) in get_step(src, NORTH)
+	STOP_PROCESSING(SSmachines,src)
+
+/obj/machinery/power/NuclearReactor/proc/CheckPipes() //In case your pipes blow up
+	outlet = null
+	outlet = locate(/obj/machinery/atmospherics/components/binary/pump) in get_step(src, NORTH)
+	if(outlet)
+		say("Success! outlet pump registered as [outlet]")
+		return TRUE
+	else
+		say("Error! No outlet pump could be found!")
+		return FALSE
+
+/obj/machinery/power/NuclearReactor/Destroy()
+	disconnect_from_network()
+	. = ..()
+
+/obj/machinery/power/NuclearReactor/proc/start()
+	if(!outlet)
+		say("ERROR! No outlet found!, please attach a standard atmospherics pump on the highlighted tile to the left!")
+	if(!powernet)
+		connect_to_network()
+	if(state & FUELHATCH_OPEN || state & WASTEHATCH_OPEN)
+		say("Warning! A hatch is still open! Please close it before trying this!")
+		return
+	say("Reaction started!")
+	Heat += 1 //kickstart it
+	START_PROCESSING(SSmachines,src)
+
+/obj/machinery/power/NuclearReactor/process()
+	if(ReactorInoperable)
+		stop()
+		return
+	UpdateIcon()
+	ProcessAtmos()
+	CheckState()
+	Time ++
+	react.ConsumeFuel()
+//	Heat = GetHeat()
+	//Efficiency = GetEfficiency() gonna play round with this temporarily
+	HeatRate = react.GetFuelAmounts()
+	Heat -= react.GetCoolAmounts()
+	if(HeatRate > 0)
+		Heat += HeatRate
+	if(Heat <= 0)
+		stop()
+		return
+
+/obj/machinery/power/NuclearReactor/proc/MeltdownAnnounce() //Announce the meltdown to give them a chance to RUN
```

Comment:
You should be fired and this proc should be placed in a hall of shame as an example of how not to code